src/library/scala/annotation/silent.scala

```diff
@@ -0,0 +1,32 @@
+/*
+ * Scala (https://www.scala-lang.org)
+ *
+ * Copyright EPFL and Lightbend, Inc.
+ *
+ * Licensed under Apache License 2.0
+ * (http://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package scala.annotation
+
+/** An annotation for local warning suppression.
+  *
+  * The optional `value` parameter allows selectively silencing messages, see `scalac -Wconf:help`
+  * for help. Examples:
+  *
+  * {{{
+  *   def f = {
+  *     1: @silent // don't warn "a pure expression does nothing in statement position"
+  *     2
+  *   }
+  *
+  *   @silent def f = { 1; deprecated() } // don't warn
+  *
+  *   @silent("msg=pure expression does nothing")
+  *   def f = { 1; deprecated() } // show deprecation warning
+  * }}}
+  */
+class silent(value: String = "") extends ConstantAnnotation
```

Comment:
I wonder if we can avoid writing the warning configuration as a raw string.
For example, I would prefer something like `@silent(msg = "deprecated")` instead of `@silent("msg=deprecated")`.
I know that one of the hurdles here is that named/default arguments don't work very well with annotations, i.e.

```
Welcome to Scala 2.12.8 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_201).
Type in expressions for evaluation. Or try :help.

scala> class annot(first: String = "", second: String = "") extends scala.annotation.Annotation
defined class annot

scala> @annot(second = "second") class Annotated
<console>:13: warning: Usage of named or default arguments transformed this annotation
constructor call into a block. The corresponding AnnotationInfo
will contain references to local values and default getters instead
of the actual argument trees
       @annot(second = "second") class Annotated
        ^
defined class Annotated
```

But maybe this is a good opportunity to fix that?

Reply:
Right, I just filed `https://github.com/scala/bug/issues/11724`. On the other hand,
  - the syntax now is exactly the same as for `-Wconf`
  - the string supports conjunctions (`cat=deprecation&origin=my.library.*`)